### PR TITLE
Maplist generator: Custom Order

### PR DESCRIPTION
**Pull request by DoubleCookies**
#297 
image
I think it is finished.
### Updates
- Created MultipleModeSelector. It looks similar to MySelect but has different handling options logic for allowing duplicates (which is a bit weird) and modes icons _(yay)_. I thought about putting everything in one file (MySelect) but with it component will be too complicated.
- Added new radio mode "Custom Order".
- Shuffling modes array is after filtering (and only in non-custom order)
- changing modes array is after getting value (without it generation starts only from second mode in array).

### CODE REVIEW

File: pages/maps.tsx [hunk 1]
```diff
@@ -347,9 +360,23 @@ const MapsGeneratorPage = () => {
...
+        options={[
+          { label: "Splat Zones", value: "SZ", data: "SZ" },
+          { label: "Tower Control", value: "TC", data: "TC" },
+          { label: "Rainmaker", value: "RM", data: "RM" },
+          { label: "Clam Blitz", value: "CB", data: "CB" },
+        ]}
+        isDisabled={generationMode !== "CUSTOM_ORDER"}
+        setValue={getModeValues}
+        width={"90%"}
+      />
```
Comment (Sendouc) [inline pages/maps.tsx:378]: This should only be rendered if Custom order is selected I think.

File: pages/maps.tsx [hunk 2]
```diff
@@ -127,15 +131,22 @@ const MapsGeneratorPage = () => {
...
+      (mode) => modeStages[mode].length > 0
+    );
+    const modes =
+      generationMode === "CUSTOM_ORDER"
+        ? filteredModes
+        : shuffleArray(filteredModes);
     if (modes.length === 0) {
-      return "I can't generate a maplist without any maps in it you know.";
+      return generationMode === "CUSTOM_ORDER"
+        ? "I can't generate a maplist without any mode you know."
```
Comment (Sendouc) [inline pages/maps.tsx:147]:
```suggestion
        ? "I can't generate a maplist without any modes you know."
```

**Review by Sendouc — CHANGES_REQUESTED** [review]
Just a few minor things but otherwise really solid work. Good job!

File: pages/maps.tsx (hunk 1 above)
Comment (DoubleCookies) [reply pages/maps.tsx]: Updated

**Pull request merged**